### ISSUE TITLE
Buttons must have discernible text

### ISSUE THREAD

**Issue by seanforyou23** [patternfly/patternfly-next#613]
Several component demos exhibit this same issue. Details related to this accessibility violation can be found here: https://dequeuniversity.com/rules/axe/3.0/button-name

Below are the URLs that illustrate the problem:
- /components/AboutModalBox/examples-full/
- /components/Button/examples-full/
- /components/Toolbar/examples-full/
- /upgrade-examples/AlertUpgradeExamples/examples-full/

This is a `critical` a11y issue.

Related to https://github.com/patternfly/patternfly-next/issues/541

To learn more about how to build and test locally, check out: https://github.com/patternfly/patternfly-next#testing-for-accessibility

**As part of this issue, please complete the following items.** These will help reviewers know what issues were addressed in the PR, and also will help us keep our total a11y violation count down.

1. Run the a11y audit, copy the issues from the audit that were fixed in the PR, and paste them into the PR description. 
    - Instructions for running the audit are here: https://github.com/patternfly/patternfly-next#testing-for-accessibility
    - Issues that are returned in the terminal look like this. Find the issues you are addressing and copy the following information provided for them or take a screencap to include in the PR.
      ```
      1 error found in: /upgrade-examples/AlertUpgradeExamples/examples-full/

      1) Buttons must have discernible text -- https://dequeuniversity.com/rules/axe/3.1/button-name -- impact: critical
      
      -------html violation instances-------

      <button class="btn btn-link dropdown-toggle" type="button" id="dropdownKebabRight" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true">
        <span class="fa fa-ellipsis-v"></span>
      </button> 
      ```
2. Reduce the threshold by number of issues being addressed 
    - This update is made [here](https://github.com/patternfly/patternfly-next/blob/master/test-a11y/config.js#L2)
    - The new value should be derived by: 
      - subtracting the number of issues you are addressing in the PR (e.g. the line that looks like this in the info you copied from the audit report in item 1 above):
        `5 errors found in: /components/Toolbar/examples-full/`
      - from current violations (e.g. this line in the audit report, near the end):
        `20 violations in total`

**Comment by jameslau** [comment]
**ISSUE**
For the third listed ARIA violation `/components/Toolbar/examples-full/` this is pointing to the fact that all buttons must have an `aria-label` embedded in the div. Below, is the violation feedback after running `npm run a11y`.

**QUESTION**
Because the buttons are dynamically generated with handlebars `pf-c-button{{#if button--modifier}} {{button--modifier}}`, is there a way to also dynamically generate the same name in the same space for the aria attributes? Say something like, `item-name--attribute='aria-label="thing"`, this would output the item name as a aria-label and then in the child element, it would generate an aria-label that matches the parent div element.

**CODE SNIPPET**
```
2 errors found in: /components/Toolbar/examples-full/

1) Elements must only use allowed ARIA attributes -- https://dequeuniversity.com/rules/axe/3.0/aria-allowed-attr -- impact: critical
2) Buttons must have discernible text -- https://dequeuniversity.com/rules/axe/3.0/button-name -- impact: critical
      
-------html violation instances-------

<button class="pf-c-options-menu__menu-item pf-m-selected" aria-selected="true">
                  Date
                  <i class="fas fa-check pf-c-options-menu__menu-item-icon" aria-hidden="true"></i>
                </button>
<button class="pf-c-options-menu__menu-item pf-m-selected" aria-selected="true">
                  Ascending
                  <i class="fas fa-check pf-c-options-menu__menu-item-icon" aria-hidden="true"></i>
                </button>
<button class="pf-c-options-menu__menu-item pf-m-selected" aria-selected="true">
                    Option 2
                    <i class="fas fa-check pf-c-options-menu__menu-item-icon" aria-hidden="true"></i>
                  </button>
<button class="pf-c-dropdown__toggle pf-m-plain" id="toolbar-simple-example-dropdown-collapsed-button" aria-expanded="false">
      <i class="fas fa-ellipsis-v" aria-hidden="true"></i></button>
<button class="pf-c-dropdown__toggle pf-m-plain" id="toolbar-pagination-example-dropdown-button" aria-expanded="false">
      <i class="fas fa-ellipsis-v" aria-hidden="true"></i></button>
<button class="pf-c-dropdown__toggle pf-m-plain" id="toolbar-expanded-example-dropdown-simple-button" aria-expanded="true">
      <i class="fas fa-ellipsis-v" aria-hidden="true"></i></button>
<button class="pf-c-dropdown__toggle pf-m-plain" id="toolbar-filter-expanded-example-dropdown-button" aria-expanded="false">
      <i class="fas fa-ellipsis-v" aria-hidden="true"></i></button>
<button class="pf-c-dropdown__toggle pf-m-plain" id="toolbar-filter-expanded-checked-example-dropdown-simple-button" aria-expanded="false">
      <i class="fas fa-ellipsis-v" aria-hidden="true"></i></button>
<button class="pf-c-dropdown__toggle pf-m-plain" id="toolbar-mobile-filter-example-dropdown-button" aria-expanded="false">
      <i class="fas fa-ellipsis-v" aria-hidden="true"></i></button>
<button class="pf-c-dropdown__toggle pf-m-plain" id="toolbar-mobile-filter-checked-example-dropdown-simple-button" aria-expanded="false">
      <i class="fas fa-ellipsis-v" aria-hidden="true"></i></button> 
```

**SUGGESTIONS**
Does anyone here (@seanforyou23 / @mcoker / @jgiardino / @dgutride) have a suggestion on how to modify the handlebars to match dynamically driven names?

**NOTES**
Besides this issue, all other aria violations has been correct. @rachael-phillips, is it possible to separate this one question as a standalone PR? Thoughts?

**Comment by jgiardino** [comment]
In the Toolbar component, this error pertains to the following buttons:
**Error:**
`1) Elements must only use allowed ARIA attributes`
**Buttons:**
```
<button class="pf-c-options-menu__menu-item pf-m-selected" aria-selected="true">
<button class="pf-c-options-menu__menu-item pf-m-selected" aria-selected="true">
<button class="pf-c-options-menu__menu-item pf-m-selected" aria-selected="true">
```
`aria-selected` isn't a valid attribute for `role="button"` (i.e. the `<button>` element), so we can just remove that attribute from these elements. As a sidenote, we will need to revisit a lot of these menu components once we have working examples in react, since the aria attributes here are closely tied with how we handle the keyboard interaction.

For the other error—`2) Buttons must have discernible text`—it looks like all of these are kebabs. When I look at the Dropdown component page, I see that the aria-label is defined in this handlebar example by including `aria-label="Actions"`:
`{{#> dropdown id="dropdown-example-kebab" dropdown--IsActionMenu="true" dropdown-toggle--modifier="pf-m-plain" dropdown--HasKebabIcon="true" aria-label="Actions"}}`

I'm guessing that you should be able to do the same for the Toolbar component. For example, right now, one of the kebabs is defined like this:
`{{#> dropdown id=(concat toolbar--id '-dropdown-collapsed') dropdown--modifier="pf-c-toolbar__action-list" dropdown--HasKebabIcon="true" dropdown-toggle--modifier="pf-m-plain"}}`
You should be able to just define `aria-label="Actions"` for these, too.

**Comment by jgiardino** [comment]
Oh, but re-reading this issue description, I think you would only need to focus on the updates related to this error: `2) Buttons must have discernible text`

I'm assuming there's a separate issue tracking this error: `1) Elements must only use allowed ARIA attributes`.

**Comment by srambach** [comment]
I still see the discernible text error on the kebabs. Is this intended to fix that or not? If not, then it probably shouldn't say that it fixes the issue because I think that will close the issue.

Also, this changes the HTML structure of a PF3 upgrade example, which isn't the point that we work with existing PF3 HTML? Also, Dana's PR to exclude updated examples was merged this morning. So we might want to leave that as is.
